pkg/console/controllers/route/controller.go

```diff
@@ -224,6 +233,24 @@ func (c *RouteSyncController) SyncCustomRoute(operatorConfig *operatorsv1.Consol
 	return customRoute, "", customRouteError
 }
 
+func (c *RouteSyncController) ValidateDefaultRouteConfig(operatorConfig *operatorsv1.Console) (*corev1.Secret, error) {
+	// if custom route is set, we don't need to validate the config
+	// since it will be used for the custom route, not the default one
+	if routesub.IsCustomRouteSet(operatorConfig) {
+		return nil, nil
+	}
+
+	if !routesub.IsCustomTLSSecretSet(operatorConfig) {
+		return nil, nil
+	}
+
+	secret, secretErr := c.secretClient.Secrets(api.OpenShiftConfigNamespace).Get(c.ctx, operatorConfig.Spec.Route.Secret.Name, metav1.GetOptions{})
+	if secretErr != nil {
+		return nil, fmt.Errorf("failed to GET default route TLS secret: %s", secretErr)
+	}
+	return secret, nil
+}
+
 func (c *RouteSyncController) ValidateCustomRouteConfig(operatorConfig *operatorsv1.Console) (*corev1.Secret, error) {
```

Comment:
Existing code, but this method is overloaded. It both gets and validates, which makes it harder to understand. We can leave this for now, but we might want to break this up into separate methods that each do one thing in the future.